client/core/trade.go

```diff
@@ -1276,6 +1276,145 @@ func (t *trackedTrade) isSwappable(ctx context.Context, match *matchTracker) (re
 	return false, false
 }
 
+// checkSwapFeeConfirms returns whether the swap fee confirmations should be
+// checked.
+//
+// This method accesses match fields and MUST be called with the trackedTrade
+// mutex lock held for reads.
+func (t *trackedTrade) checkSwapFeeConfirms(match *matchTracker) bool {
+	if match.MetaData.Proof.SwapFeeConfirmed {
+		return false
+	}
+	_, dynamic := t.wallets.fromWallet.Wallet.(asset.DynamicSwapOrRedemptionFeeChecker)
+	if !dynamic {
+		// Confirmed will be set in the db.
+		return true
+	}
+	if match.Side == order.Maker {
+		if match.Status < order.MakerSwapCast {
+			return false
+		}
+	} else {
+		if match.Status < order.TakerSwapCast {
+			return false
+		}
+	}
+	return true
+}
+
+// checkRedemptionFeeConfirms returns whether the swap fee confirmations should
+// be checked.
+//
+// This method accesses match fields and MUST be called with the trackedTrade
+// mutex lock held for reads.
+func (t *trackedTrade) checkRedemptionFeeConfirms(match *matchTracker) bool {
+	if match.MetaData.Proof.RedemptionFeeConfirmed {
+		return false
+	}
+	_, dynamic := t.wallets.toWallet.Wallet.(asset.DynamicSwapOrRedemptionFeeChecker)
+	if !dynamic {
+		// Confirmed will be set in the db.
+		return true
+	}
```

Comment:
``` golang
if (match.Side == order.Maker && match.Status < order.MakerRedeemed) || match.Status < order.MatchComplete {
    return false
}
```

Reply:
Can shorted below? Will change.

Reply:
See related suggestion in https://github.com/decred/dcrdex/pull/1752/files#r974331842
We should be returning the result of an expression rather than boolean literals to simplify these.